core/authenticator_factory.go

```diff
@@ -61,7 +61,12 @@ func GetAuthenticatorFromEnvironment(credentialKey string) (authenticator Authen
 	} else if strings.EqualFold(authType, AUTHTYPE_NOAUTH) {
 		authenticator, err = NewNoAuthAuthenticator()
 	} else {
-		err = fmt.Errorf(ERRORMSG_AUTHTYPE_UNKNOWN, authType)
+		err = coreSDKErrorf(
+			nil,
+			fmt.Sprintf(ERRORMSG_AUTHTYPE_UNKNOWN, authType),
+			"unknown-auth-type",
```

Comment:
We could, but they are not necessarily meant to be reused and in fact, it doesn't matter if they are re-used or not. They are just meant to force uniqueness between error hashes that would otherwise be the same (same function, etc.)